app/controllers/react_controller.rb

```diff
@@ -4,6 +4,6 @@ class ReactController < ApplicationController
 
   def index
     response.headers['X-Request-Path'] = request.path
-    render 'react/index'
+    render 'react/index.html.erb'
```

Comment:
Clearly the test failures are related, so I'd say the current patch is invalid. Based on https://guides.rubyonrails.org/layouts_and_rendering.html I'd try this to force the HTML format:
```suggestion
    render html: 'react/index'
```

Reply:
As far as I understand "render html:"  is used to render html string directly. https://guides.rubyonrails.org/layouts_and_rendering.html#rendering-html
```
    render html: 'react/index'
```


Are those tests failure due to the deprecated warning or something else are failed?

Reply:
They are because of the deprecation. We make those fatal in ci and I haven't seen them before

This really is an edge case, since the ics tld doesn't exist

Can the controller force the format in another way?

Reply:
I tried the below before suggesting the change in this PR, but none of them worked:
1. Pass ":format => 'html' to the 'constraints'
2. Set `request.format = 'html' in the react controller

Reply:
Oh, changing the `request.format = 'html'` in "before_action" appears to work.

```
class ReactController < ApplicationController
  layout 'layouts/react_application'
  skip_before_action :authorize, :only => :index
  before_action :force_html_format

  def force_html_format
    request.format = 'html'
  end

  def index
    response.headers['X-Request-Path'] = request.path
    render 'react/index'
  end
end
```

Reply:
```suggestion
    render("react/index", formats: [:html])
```
This works for me too.

Reply:
Used what @swimlappy used marking this as resolved.